plugins/history_plugin/history_plugin.cpp

```diff
@@ -438,6 +448,42 @@ namespace eosio {
         return result;
       }
 
+      read_write::prune_history_result read_write::prune_history( const read_write::prune_history_params& params ) const {
+         auto& chain = history->chain_plug->chain();
+         chainbase::database& db = const_cast<chainbase::database&>( chain.db() ); // Override read-only access to state DB (highly unrecommended practice!)
+
+         prune_history_result result{-1, false};
+
+         if(chain.irreversible_read_mode()) {
+            auto start_time = fc::time_point::now();
+
+            const auto& idx = db.get_index<action_history_index, by_id>();
+            auto itr = idx.lower_bound(history->last_id_pruned);
+
+            while( itr != idx.end() &&  itr->block_num <= params.height ) {
+               fc::datastream<const char*> ds( itr->packed_action_trace.data(), itr->packed_action_trace.size() );
+               action_trace t;
+               fc::raw::unpack( ds, t );
+
+               auto &a = db.get<account_history_object, by_account_action_seq>(boost::make_tuple(t.receipt->receiver, t.receipt->recv_sequence));
```

Comment:
Looks like you need to add an index to `account_history_object` for `account_sequence_num` and use that to find and remove. I know we talked about not adding an index, but I don't see any way around it. Anyone not running with the greymass patch for history_plugin will need to do a replay or start from a given block via a snapshot anyway. The `account_sequence_num` index on `account_history_object` will need to fold in `id` so that it is `ordered_unique`. See large comment in `chain/types.hpp` for details.

The good news about that is unpack of the `packed_action_trace` will no longer be needed.